resource-watch/doc-api
# Updates to the query endpoint documentation

## Summary

This PR refers to [GTC-324](https://gfw.atlassian.net/browse/GTC-324). It represents an update on the documentation of the endpoints for querying datasets (`query`),  downloading query results (`download`), and getting the possible charts that one can build using a given query (`jiminy`).

## Testing instructions

If you feel like deep-diving into the contents of the PR, follow the steps below:

1. Clone the documentation repository and follow the [installation instructions on the README](https://github.com/resource-watch/doc-api#installation-and-usage).
2. Checkout the branch of this PR: `docs/query-endpoints`
3. Read the updated docs for the **query endpoint**, **download endpoint**, and **jiminy endpoint**. 
4. Check also the added sections for SQL compatibility for **CartoDB datasets** and **document-based datasets**, as well as the new developer docs section about **query transformations**. 

*The following SQL features not being covered by this version of the docs:*

* Selecting using PostGIS functions
  * SELECT ST_METADATA(rast)
  * SELECT ST_BANDMETADATA(rast, occurrence)
  * SELECT ST_SUMMARYSTATS()
  * SELECT ST_HISTOGRAM(rast, 1, auto, true)
  * SELECT ST_valueCount(rast, 1, true)
* Where conditions using PostGIS functions
* Grouping by SQL functions OR PostGIS functions
* Document-based dataset custom group by functions: `date_histogram(field='insert_time','interval'='1d')` and `date_range(field='insert_time','format'='yyyy-MM-dd' ,'2014-08-18','2014-08-17','now-8d','now-7d','now-6d','now')`
* Ordering by SQL functions OR PostGIS functions
* Analysis of `geostore` query parameter and its influence on the query results.
